src/main/java/it/pagopa/transactions/utils/TransactionsUtils.java

```diff
@@ -10,14 +10,42 @@
 import org.springframework.stereotype.Component;
 import reactor.core.publisher.Mono;
 
+import java.util.EnumMap;
+import java.util.Map;
+
 @Component
 public class TransactionsUtils {
 
     private final TransactionsEventStoreRepository<Object> eventStoreRepository;
 
+    private final Map<it.pagopa.ecommerce.commons.generated.server.model.TransactionStatusDto, it.pagopa.generated.transactions.server.model.TransactionStatusDto> transactionStatusLookupMap = new EnumMap<>(
+            it.pagopa.ecommerce.commons.generated.server.model.TransactionStatusDto.class
+    );
+
     @Autowired
     public TransactionsUtils(TransactionsEventStoreRepository<Object> eventStoreRepository) {
         this.eventStoreRepository = eventStoreRepository;
+        init();
+    }
+
+    private void init() {
+        for (it.pagopa.ecommerce.commons.generated.server.model.TransactionStatusDto enumValue : it.pagopa.ecommerce.commons.generated.server.model.TransactionStatusDto
+                .values()) {
+            /*
+             * @formatter:off
+             * This lookup map serve two purpose:
+             * 1) handle enumerations conversion from commons and transactions-service TransactionStatusDto enumerations
+             * 2) in case one enumeration from commons is not present into transactions-service's ones
+             * then an IllegalArgumentException is raised, preventing the module from being
+             * run and avoiding runtime errors correlated to specs updated into commons
+             * that are not reflected into transactions specs (such as a transaction status added only into commons)
+             * @formatter:on
+             */
+            transactionStatusLookupMap.put(
+                    enumValue,
+                    it.pagopa.generated.transactions.server.model.TransactionStatusDto.fromValue(enumValue.toString())
+            );
```

Comment:
I would add something like
```suggestion
            );
            
            if (it.pagopa.ecommerce.commons.generated.server.model.TransactionStatusDto.values().length != it.pagopa.generated.transactions.server.model.TransactionStatusDto.values().length) {
                String commonsValues = Arrays.toString(it.pagopa.ecommerce.commons.generated.server.model.TransactionStatusDto.values);
                String transactionValues = Arrays.toString(it.pagopa.generated.transactions.server.model.TransactionStatusDto.values);
                throw new IllegalArgumentException("Mismatched transaction status enumerations: (commons)\n" + commonsValues + "\n\nvs\n\n(transactions)\n" + transactionValues)
            }
```

So that if there is a value in transactions that is not present in commons an exception is raised too

Reply:
Good catch.

Fixed with https://github.com/pagopa/pagopa-ecommerce-transactions-service/pull/169/commits/d06f51375cab419c5e9049e85e71b60c0f582244

Now a message is thrown with the unhandled statuses for commons and transactions so that only the mismatching statuses are reported into exception message for make it easier to find the mismatching statuses from common and/or transactions enumerations